x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/NlpTaskConfig.java

```diff
@@ -56,15 +60,18 @@ public static String documentId(String model) {
     private final boolean doLowerCase;
     private final boolean withSpecialTokens;
     private final List<String> classificationLabels;
+    private final Integer maxSequenceLength;
 
     NlpTaskConfig(TaskType taskType, List<String> vocabulary,
                   boolean doLowerCase, boolean withSpecialTokens,
-                  List<String> classificationLabels) {
+                  List<String> classificationLabels,
+                  Integer maxSequenceLen) {
         this.taskType = taskType;
         this.vocabulary = vocabulary;
         this.doLowerCase = doLowerCase;
         this.withSpecialTokens = withSpecialTokens;
         this.classificationLabels = classificationLabels;
+        this.maxSequenceLength = maxSequenceLen;
```

Comment:
Shall we set the default value here if the user did not set it? Then the parameter value will be visible and all callers won't have to worry about `null`.

Reply:
The default should be 512 for BERT models.

Reply:
Different models will have different defaults. For BERT flavour models 512 is a good place to start although that value can differ between models. For a non BERT model the default could be something else. The task config class is model agnostic so it knows nothing about the features of individual models, the defaulting logic is further down the pipeline in `BertRequestBuilder` where the domain specific knowledge is encoded.

Reply:
Should we also add this to the HuggingFace upload script? To grab the input size from the config file?

Reply:
I changed my mind about this while eating lunch:

- A default is required by all models so there is no point coding an abstraction this is not used
- `NlpTaskConfig` is already tied to BERT via the `BertTokenizer buildTokenizer()` method
- Adding the default to the config means the field is written out in the serialised doc making the option visible and documented

I pushed a change to default the value in  `NlpTaskConfig`